Jenkinsfile

```diff
@@ -146,17 +145,25 @@ node('docker && imgtec') {  // Only run on internal slaves as build takes a lot
             sh 'make defconfig'
         }
         stage('Build') {
-            // Attempt to build quickly and reliably
-            try {
-                sh "make -j4 V=s ${params.ALL_PACKAGES ? 'IGNORE_ERRORS=m' : ''}"
-            } catch (hudson.AbortException err) {
-                // TODO BUG JENKINS-28822
-                if(err.getMessage().contains('script returned exit code 143')) {
-                    throw err
+            // Add opkg signing key
+            withCredentials([
+                [$class: 'FileBinding', credentialsId: 'opkg-build-private-key', variable: 'PRIVATE_KEY'],
+                [$class: 'FileBinding', credentialsId: 'opkg-build-public-key', variable: 'PUBLIC_KEY'],
+            ]){
+                sh "ln -s ${env.PRIVATE_KEY} ${WORKSPACE}/key-build"
+                sh "ln -s ${env.PUBLIC_KEY} ${WORKSPACE}/key-build.pub"
+                // Attempt to build quickly and reliably
+                try {
+                    sh "make -j4 V=s ${params.ALL_PACKAGES ? 'IGNORE_ERRORS=m' : ''}"
+                } catch (hudson.AbortException err) {
+                    // TODO BUG JENKINS-28822
+                    if(err.getMessage().contains('script returned exit code 143')) {
+                        throw err
+                    }
+                    echo 'Parallel build failed, attempting to continue in  single threaded mode'
                 }
-                echo 'Parallel build failed, attempting to continue in  single threaded mode'
+                sh "make -j1 V=s ${params.ALL_PACKAGES ? 'IGNORE_ERRORS=m' : ''}"
```

Comment:
Seeing as you are changing all this could you move the make inside the catch, just after echo. As that is where it should be. Otherwise technically the build runs twice when there is no failure.